presto-main/src/main/java/com/facebook/presto/operator/GroupByHash.java

```diff
@@ -64,4 +64,6 @@ static GroupByHash createGroupByHash(
     boolean contains(int position, Page page, int[] hashChannels);
 
     int putIfAbsent(int position, Page page);
+
+    int compare(int leftGroupId, int rightGroupId);
```

Comment:
You don't need to double memory usage. You could spill some constant number of pages at a time.

`compare` seems very out of place since it will fail for types that are not orderable, whereas all the other operations work. Will discuss with a couple other people to get their opinions.

Another option might be to add an API that allows a lambda to be applied to a group id.